### PR TITLE
Add a docstring for maps.Mesh

### PR DESCRIPTION
**Pull request by yashk2810**

### CODE REVIEW

File: jax/interpreters/pxla.py [hunk 1]
```diff
@@ -1832,6 +1832,49 @@ def _pmap_lowering(ctx, *in_nodes, axis_name,
 # ------------------- xmap -------------------
 
 class Mesh(ContextDecorator):
+  """Declare the hardware resources available in the scope of this manager.
+
+  In particular, all ``axis_names`` become valid resource names inside the
+  managed block and can be used e.g. in the ``axis_resources`` argument of
+  :py:func:`xmap`.
+
+  If you are compiling in multiple threads, make sure that the
```
Comment (hawkinsp) [inline jax/interpreters/pxla.py:1841]: I wonder if this is worth including. Is there ever a context manager that works any other way?

File: jax/interpreters/pxla.py [hunk 2]
```diff
@@ -1832,6 +1832,49 @@ def _pmap_lowering(ctx, *in_nodes, axis_name,
 # ------------------- xmap -------------------
 
 class Mesh(ContextDecorator):
+  """Declare the hardware resources available in the scope of this manager.
+
+  In particular, all ``axis_names`` become valid resource names inside the
+  managed block and can be used e.g. in the ``axis_resources`` argument of
+  :py:func:`xmap`.
```
Comment (hawkinsp) [inline jax/interpreters/pxla.py:1839]: Cross-reference to the user-friendly introduction to these concepts?

File: jax/interpreters/pxla.py [hunk 3]
```diff
@@ -1832,6 +1832,49 @@ def _pmap_lowering(ctx, *in_nodes, axis_name,
...
+  Example:
+
+    >>> from jax.experimental.maps import Mesh
+    >>> from jax.experimental.pjit import pjit
+    >>> from jax.experimental import PartitionSpec as P
+    >>> import numpy as np
+    ...
+    >>> inp = np.arange(16).reshape((8, 2))
+    >>> devices = np.array(jax.devices()).reshape(4, 2)
+    ...g
```
Comment (hawkinsp) [inline jax/interpreters/pxla.py:1861]: Typo `g` ?

**Review by hawkinsp — CHANGES_REQUESTED** [review]
Please get Jake's review also.

File: jax/interpreters/pxla.py (hunk 2 above)
Comment (yashk2810) [reply jax/interpreters/pxla.py:1839]: Done. Linked to pjit and multi-process doc.

File: jax/interpreters/pxla.py (hunk 1 above)
Comment (yashk2810) [reply jax/interpreters/pxla.py:1841]: It lead to some confusion before, so I think its fine to include this.

File: jax/interpreters/pxla.py (hunk 3 above)
Comment (yashk2810) [reply jax/interpreters/pxla.py:1861]: Done.

**Review by jakevdp — APPROVED** [review]
Looks great!

**Pull request merged**